src/web/src/components/Header.vue

```diff
@@ -1,109 +1,87 @@
-        <div>
-          <b-button v-b-modal.login-modal size="sm" variant="light">
-            Log In
-          </b-button>
-
-          <b-button
-            v-b-modal.signup-modal
-            size="sm"
-            variant="primary"
-            class="ml-2"
-          >
-            Sign Up
-          </b-button>
-
-          <b-modal
-            id="login-modal"
-            ref="login-modal"
-            hide-footer
-            title="Log In"
-          >
-            <b-form @submit="onSubmit" @reset="onReset" v-if="showForm">
-              <b-form-group
-                id="input-group-1"
-                label="Email address:"
-                label-for="input-1"
-              >
-                <b-form-input
-                  id="input-1"
-                  v-model="form.email"
-                  type="email"
-                  required
-                  placeholder="Enter email"
-                ></b-form-input>
-              </b-form-group>
-
-              <b-form-group
-                id="input-group-2"
-                label="Password:"
-                label-for="input-2"
-              >
-                <b-form-input
-                  id="input-2"
-                  type="password"
-                  v-model="form.password"
-                  required
-                  placeholder="Enter password"
-                ></b-form-input>
-              </b-form-group>
-
-              <b-button type="submit" variant="primary">Submit</b-button>
-            </b-form>
-          </b-modal>
-
-          <b-modal id="signup-modal" hide-footer title="Sign Up">
-            <SignUpForm></SignUpForm>
-          </b-modal>
-        </div>
-      </b-navbar-nav>
-      <b-form-checkbox
-        class="dark-mode-switch"
-        :checked="$store.state.darkMode"
-        @change="toggle_style()"
-        switch
-      >
-        <font-awesome-icon class="style-icon" :icon="faMoon" />
-      </b-form-checkbox>
+      <b-nav-text class="semester">{{ selectedSemester }}</b-nav-text>
+      <b-navbar-toggle target="header-navbar-collapse"></b-navbar-toggle>
+      <b-collapse id="header-navbar-collapse" is-nav>
+        <b-navbar-nav>
+          <b-nav-item>
+            <router-link :to="{ name: 'CourseScheduler' }">
+              <font-awesome-icon :icon="faCalendar" />
+              Schedule
+            </router-link>
+          </b-nav-item>
+          <b-nav-item>
+            <router-link :to="{ name: 'CourseExplorer' }">
+              <font-awesome-icon :icon="faList" />
+              Explore
+            </router-link>
+          </b-nav-item>
+        </b-navbar-nav>
+        <!-- If user has logged in -->
+        <b-navbar-nav class="ml-auto">
+          <b-nav-item-dropdown right v-if="sessionID !== null">
+            <!-- Using 'button-content' slot -->
+            <template v-slot:button-content>Hi, {{ userName }}</template>
+            <!-- <b-dropdown-item href="#">Profile</b-dropdown-item> -->
```

Comment:
Please remove unused code comments, lines 25,27,33.
Besides that lgtm!

Reply:
Removed